catroid/src/main/java/org/catrobat/catroid/utils/TextSizeUtil.java

```diff
@@ -184,33 +184,52 @@ public static void enlargeOptionsItem(MenuItem item) {
 		item.setTitle(spanString);
 	}
 
-	public static void enlargePreference(PreferenceActivity prefrenceActivity, String key) {
-		Preference preference = (Preference) prefrenceActivity.findPreference(key);
+	public static void enlargePreferenceGroup(PreferenceGroup preferenceGroup) {
+		if (modifier == 1.0f) {
+			return;
+		}
+
+		for (int i = 0; i < preferenceGroup.getPreferenceCount(); i++) {
+			Preference preference = preferenceGroup.getPreference(i);
+			if (preference instanceof PreferenceScreen) {
+				enlargePreferenceScreen((PreferenceScreen) preference);
+				enlargePreferenceGroup((PreferenceGroup) preference);
+			} else if (preference instanceof PreferenceCategory) {
+				enlargePreferenceCategory((PreferenceCategory) preference);
+				enlargePreferenceGroup((PreferenceGroup) preference);
+			} else if (preference instanceof CheckBoxPreference) {
+				enlargeCheckBoxPreference((CheckBoxPreference) preference);
+			} else if (preference instanceof ListPreference) {
+				enlargeListPreference((ListPreference) preference);
+			} else if (preference instanceof EditTextPreference) {
+				enlargeEditTextPreference((EditTextPreference) preference);
+			} else if (preference instanceof Preference) {
+				enlargePreference(preference);
+			}
+		}
+	}
+
+	public static void enlargePreference(Preference preference) {
 		preference.setLayoutResource(R.layout.preference_enlarged);
```

Comment:
can those be combined into one method since you basically call the same method `setLayoutResource(R.layout.preference_enlarged)` in all of them?

Reply:
Good point! Changed it.